modules/tinymce/src/plugins/link/main/ts/core/Actions.ts

```diff
@@ -78,6 +82,10 @@ const createLink = (editor: Editor, selectedElm: Element, text: Optional<string>
     text.fold(
       () => {
         editor.execCommand('mceInsertLink', false, linkAttrs);
+        // Now the newly inserted link is selected. Move the cursor behind the new link, so the user can go on typing.
+        const rng = dom.createRng();
+        rng.setStartAfter(editor.selection.getEnd());
+        editor.selection.setRng(rng);
```

Comment:
Just to explicitly make sure the new rng is collapsed
```suggestion
        const end = editor.selection.getEnd();
        const rng = dom.createRng();
        rng.setStartAfter(end);
        rng.setEndAfter(end);

```

Reply:
Did you intentionally delete the `editor.selection.setRng(rng);` line? I've added it back.

Reply:
That was an accident, thanks for picking that up